vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go

```diff
@@ -194,18 +197,17 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 }
 
 func (r *ReconcileMachine) getCluster(ctx context.Context, machine *clusterv1.Machine) (*clusterv1.Cluster, error) {
+	if machine.Labels[MachineClusterLabelName] == "" {
+		klog.Infof("Machine %q in namespace %q doesn't specify %q label, assuming nil cluster", machine.Name, MachineClusterLabelName, machine.Namespace)
+		return nil, nil
+	}
+
 	clusterList := clusterv1.ClusterList{}
 	listOptions := &client.ListOptions{
-		Namespace: machine.Namespace,
-		// This is set so the fake client can be used for unit test. See:
-		// https://github.com/kubernetes-sigs/controller-runtime/issues/168
-		Raw: &metav1.ListOptions{
-			TypeMeta: metav1.TypeMeta{
-				APIVersion: clusterv1.SchemeGroupVersion.String(),
-				Kind:       "Cluster",
-			},
-		},
+		FieldSelector: fields.OneTermEqualSelector("metadata.name", machine.Labels[MachineClusterLabelName]),
```

Comment:
@vincepri, got the error: Index with name field:metadata.name does not exist.
it should be `fields.OneTermEqualSelector("metadata.name", machine.Labels[MachineClusterLabelName]).String()`?